src/resolver.js

```diff
@@ -70,19 +70,33 @@ function resolve(link, ctx, visited = []) {
         ctx.result.push(createYAMLParseError(e, ctx, resolvedFilePath));
         return { node: undefined };
       }
-      // FIXME: lost yaml parsing and file read errors here
     } else if (isFullyQualifiedUrl(resolvedFilePath)) {
       try {
-        const xhr = new XMLHttpRequest();
-        xhr.open('GET', resolvedFilePath, false);
-        xhr.send();
+        ctx.redoclyClient.processRegistryDependency(resolvedFilePath, ctx);
```

Comment:
also, can we do it in custom rule vs here in resolver? e.g. in `semantic/registryDependencies.js`